visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -221,98 +195,45 @@ export class CodeMapMouseEventService
 				const to = this.intersectedBuilding ? this.intersectedBuilding : this.highlightedInTreeView
 
 				if (from !== to) {
+					if (this.temporaryLabelForBuilding !== null) {
+						this.codeMapLabelService.clearTemporaryLabel(this.temporaryLabelForBuilding)
+						this.temporaryLabelForBuilding = null
+					}
+
+					this.threeSceneService.resetLabel()
 					this.unhoverBuilding()
 					if (to) {
+						if (to.node.isLeaf) {
+							let labelForBuilding = this.threeSceneService.getLabelForHoveredNode(to, labels)
+							labelForBuilding = labelForBuilding !== null ? labelForBuilding : this.drawTemporaryLabel(to, labels)
+							this.threeSceneService.hoverLabel(labelForBuilding, this.raycaster, labels)
+						}
 						this.hoverBuilding(to)
 					}
 				}
 			}
 		}
 	}
 
-	private isOverlapping1D(minBox1: number, maxBox1: number, minBox2: number, maxBox2: number) {
-		return maxBox1 >= minBox2 && maxBox2 >= minBox1
-	}
-
-	getIntersectionDistance(bboxHoveredLabel: Box3, bboxObstructingLabel: Box3, normedVector: Vector3, distance: number) {
-		normedVector.multiplyScalar(distance)
-		bboxHoveredLabel.translate(normedVector)
-
-		if (
-			(this.isOverlapping1D(bboxObstructingLabel.min.x, bboxObstructingLabel.max.x, bboxHoveredLabel.min.x, bboxHoveredLabel.max.x) &&
-				this.isOverlapping1D(
-					bboxObstructingLabel.min.y,
-					bboxObstructingLabel.max.y,
-					bboxHoveredLabel.min.y,
-					bboxHoveredLabel.max.y
-				)) ||
-			(this.isOverlapping1D(bboxObstructingLabel.min.x, bboxObstructingLabel.max.x, bboxHoveredLabel.min.x, bboxHoveredLabel.max.x) &&
-				this.isOverlapping1D(
-					bboxObstructingLabel.min.z,
-					bboxObstructingLabel.max.z,
-					bboxHoveredLabel.min.z,
-					bboxHoveredLabel.max.z
-				)) ||
-			(this.isOverlapping1D(bboxObstructingLabel.min.y, bboxObstructingLabel.max.y, bboxHoveredLabel.min.y, bboxHoveredLabel.max.y) &&
-				this.isOverlapping1D(
-					bboxObstructingLabel.min.z,
-					bboxObstructingLabel.max.z,
-					bboxHoveredLabel.min.z,
-					bboxHoveredLabel.max.z
-				))
-		) {
-			return distance
-		}
-		return 0
-	}
+	private drawTemporaryLabel(to: CodeMapBuilding, labels: Object3D[]) {
```

Comment:
We could rename variabel `to` and the method name:
```suggestion
	private drawTemporaryLabelFor(building: CodeMapBuilding, labels: Object3D[]) {
```